src/realm/sync/noinst/migration_store.cpp

```diff
@@ -305,4 +311,31 @@ void MigrationStore::create_subscriptions(const SubscriptionStore& subs_store, c
     mut_sub.commit();
 }
 
+void MigrationStore::create_sentinel_subscription_set(const SubscriptionStore& subs_store)
+{
+    std::lock_guard lock{m_mutex};
+    if (m_state != MigrationState::Migrated) {
+        return;
+    }
+    if (m_sentinel_subscription_set_version) {
+        return;
+    }
+    auto mut_sub = subs_store.get_latest().make_mutable_copy();
+    auto subscription_set_version = mut_sub.commit().version();
+    m_sentinel_subscription_set_version.emplace(subscription_set_version);
+
+    auto tr = m_db->start_write();
+    auto migration_table = tr->get_table(m_migration_table);
+    REALM_ASSERT(!migration_table->is_empty());
+    auto migration_store_obj = migration_table->get_object(0);
+    migration_store_obj.set(m_sentinel_query_version, *m_sentinel_subscription_set_version);
```

Comment:
I think this is actually all you need? Just a write that sets this flag and then you can use the snapshot version of that write as a limit the way we use the snapshot version of a subscription set as a limit. Do we actually need to create a new subscription set?

Reply:
A flag would not be enough. We would need the snapshot version in case the session gets closed before uploading everything. This is the reason the sentinel value is stored.